### PR TITLE
Homework3

### PR DESCRIPTION
**Pull request by Karl5236543**

### CODE REVIEW

File: homework3.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    #---------------------------------------------------#
+    # 2
+    #---------------------------------------------------#
+    print("---------3----------")
+    sum3(12,10)
+    sum3(1,3)
+    sum3(12,10)
+    sum3(12,10)
+    sum3(a=12,b=10)
+    sum3(a=12,b=10)
```
Comment (katevoskresenska) [inline homework3.py:85]: Looks good!

File: homework3.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+def divided_by_one_hundred(func):
+    @functools.wraps(func)
+    def inner(*args, **kwargs):
+        res = func(*args, **kwargs)
+        if res % 100 == 0:
+            print("We are OK!")
+        else:
+            print(f"Bad news guys, we got {res % 100}")
+        return res
+    return inner
```
Comment (katevoskresenska) [inline homework3.py:13]: If I am not mistaken, the task was to check if 100 can be divided by the number without remainder of the division

File: homework3.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    def inner(*args, **kwargs):
+        if True in [isinstance(el, str) for el in (args + tuple(kwargs.values()))]:
+            raise ValueError("string type is not supported")
+        return func(*args, **kwargs)
+    return inner
+
+
+def cached(func):
+    cache = {}
+    used_cache_count = 0
```
Comment (mmykhailenko) [inline homework3.py]: `inner. used_cache_count = 0 `it gives you chance to call this attr when you will call your function `sum3 . used_cache_count ` and also no need then `nonlocal`

File: homework3.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        else:
+            print(f"Bad news guys, we got {100 % res}")
+        return res
+    return inner
+
+
+def not_str(func):
+    @functools.wraps(func)
+    def inner(*args, **kwargs):
+        if True in [isinstance(el, str) for el in (args + tuple(kwargs.values()))]:
```
Comment (mmykhailenko) [inline homework3.py]: check the `any()` or `all()` functions in Python, they will bring you more clear code

**Review by mmykhailenko — APPROVED** [review]
LGTM

**Pull request merged**